### PR TITLE
fix(diffusion): DiffuGRPO TRL buffering, smoke example, [grpo] extra

### PR DESCRIPTION
**Pull request by nishide-dev**
## Summary

- **DiffuGRPOTrainer**: Fix training with current TRL `split_tensor_dict` buffering by keeping `mask_seeds` on the trainer (lists were being sliced to empty), skipping `shuffle_sequence_dict` so rows stay aligned with cached log-probs, matching the approach in dllm/d1.
- **Compatibility**: `epsilon_low` / `epsilon_high` for clipping; `warnings_issued` init for in-memory `PreTrainedModel`; `torch.amp.autocast("cuda")` instead of deprecated `torch.cuda.amp.autocast`.
- **Example**: `examples/grpo_diffu_train_smoke.py` — GPU smoke for a few `train()` steps (`import unsloth` first; optional TRL import stubs for weave/mergekit/judges when absent).
- **Packaging**: `pyproject` optional extra `[grpo]` (`trl`, `mergekit`, `tokenizers`); README, Mintlify docs (EN/JA), CLAUDE.md, CONTRIBUTING updated.
- **Docs**: Correct `_forward_process` docstring (d1 CE path does not re-weight by `p_mask`).

## Testing

- `pytest tests/diffusion/test_grpo_trainer.py tests/diffusion/test_integration.py -m "not slow"`
- `python examples/grpo_diffu_train_smoke.py` (CUDA)

Refs: #226

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by nishide-dev** [comment]
## Review follow-up (automated + manual)

**`_step` in `_prepare_inputs`:** Kept `self._step += 1` — it matches upstream `trl.trainer.grpo_trainer.GRPOTrainer._prepare_inputs` (single increment per micro-batch before `compute_loss`). No second increment in `training_step` for this counter in TRL 0.24.

**Change since opening:** `torch.cuda.empty_cache()` in `_get_per_token_logps` and `_generate_and_score_completions` is now gated on `device.type == "cuda"` (same pattern as `generate()`).

Residual risk: smoke uses `num_iterations=1`; broader configs would benefit from a future test with `num_iterations > 1` and `steps_per_generation > 1`.
